packages/graphql/tests/e2e/subscriptions/single-instance-plugin.e2e.test.ts

```diff
@@ -0,0 +1,174 @@
+            },
+        });
+
+        server = new ApolloTestServer(neoSchema);
+        await server.start();
+    });
+
+    beforeEach(() => {
+        wsClient = new WebSocketTestClient(server.wsPath);
+        wsClient2 = new WebSocketTestClient(server.wsPath);
+    });
+
+    afterEach(async () => {
+        await wsClient.close();
+        await wsClient2.close();
+    });
+
+    afterAll(async () => {
+        await server.close();
+        await driver.close();
+    });
+
+    // NOTE: This test **may** be flaky, if so, feel free to remove it
+    test("listeners are properly triggered and cleaned up", async () => {
+        expect(plugin.events.getMaxListeners()).toBe(0);
+        expect(plugin.events.listenerCount("create")).toBe(0);
+
+        await wsClient.subscribe(subscriptionQuery);
+        await wsClient2.subscribe(subscriptionQuery);
+
+        await delay(50); // Sorry listener count takes a bit to update
+        expect(plugin.events.listenerCount("create")).toBe(2);
+
+        await wsClient2.close();
+        await delay(50); // Sorry listener count takes a bit to update
+        expect(plugin.events.listenerCount("create")).toBe(1);
+    });
+
+    test("multiple listeners attached to local event plugin", async () => {
+        await wsClient.subscribe(subscriptionQuery);
+        await wsClient2.subscribe(subscriptionQuery);
+
+        await createMovie("movie1");
+
+        expect(wsClient.errors).toEqual([]);
+        expect(wsClient.events).toEqual([
+            {
+                [typeMovie.operations.subscribe.created]: {
+                    [typeMovie.operations.subscribe.payload.created]: { title: "movie1" },
+                    event: "CREATE",
+                    timestamp: expect.any(Number),
+                },
+            },
+        ]);
+        expect(wsClient2.errors).toEqual([]);
+        expect(wsClient2.events).toEqual([
+            {
+                [typeMovie.operations.subscribe.created]: {
+                    [typeMovie.operations.subscribe.payload.created]: { title: "movie1" },
+                    event: "CREATE",
+                    timestamp: expect.any(Number),
+                },
+            },
+        ]);
+    });
+
+    test("create and triggers subscription", async () => {
+        await wsClient.subscribe(subscriptionQuery);
+
+        await createMovie("movie1");
+        await createMovie("movie2");
+
+        expect(wsClient.errors).toEqual([]);
+        expect(wsClient.events).toEqual([
+            {
+                [typeMovie.operations.subscribe.created]: {
+                    [typeMovie.operations.subscribe.payload.created]: { title: "movie1" },
+                    event: "CREATE",
+                    timestamp: expect.any(Number),
+                },
+            },
+            {
+                [typeMovie.operations.subscribe.created]: {
+                    [typeMovie.fieldNames.subscriptions.created]: { title: "movie2" },
```

Comment:
question: Shouldn't this also be `[typeMovie.operations.subscribe.payload.created]: `?

Reply:
Good question. mmm, I think we have a bit of duplicate code here. Both seem to be exactly the same. I think the alternative you propose is better, so I will remove the "fieldNames" property and change all references